### ISSUE TITLE
Support for logarithmic units

### ISSUE THREAD

**Issue by MatthieuDartiailh** [hgrecco/pint#249]
The idea would be to support units like dBm for power and the like. dBm are for example often used in microwave sources. I don't know if tackling is possible but I would like your opinion.

**Comment by hgrecco** [comment]
Yes. This is indeed desired feature. We started to make this route available when we introduced the non multiplicative converters. But we started with offset units (temperatures) as the demand was higher. Take a look at #71 for some initial comments about this (but Pint has evolved since then)

**Comment by joshvillbrandt** [comment]
+1

Is this functionality available by way of converters as mentioned in #71? I'm trying to manually construct a  `UnitDefinition` with a custom `Converter`, but I'm not having luck since neither of these items are exposed outside of the pint library.

My goal is to convert between [AWG](http://en.wikipedia.org/wiki/American_wire_gauge) and a standard unit of length (mm in this case.) The conversion function I'd like to add is `
american_wire_gauge = 8.62491 * log(1/mm) + 18.2019 = AWG = awg`, but of course `log` isn't available. I think adding support for logs directly is out of my league. Alternatively, I can fork pint and bake the conversion directly into the library, but is a custom AWG converter something that would/should be merged into the primary repo here?

**Comment by hgrecco** [comment]
@joshvillbrandt Thanks for looking into this. I will be happy to assist you. 

Even if they are not exposed, you can still access. For example
```python
from pint.unit import UnitDefinition, Converter
class LogConverter(Converter):
    # your stuff here
```
and then test it:
```python
ureg = pint.UnitRegistry()
ureg.define(my_definition)
```
where `my_definition` is a `UnitDefinition` that uses your `LogConverter`

Apart from this, you can fork it and then do a PR. It will be very much welcome. Just do the following:
1.- Fork the develop branch
2.- Add your new converter to converters.py
3.- Add test cases in which you test the converter independently
4.- Add test cases in which you test the converter by adding programmatically a unit to the registry.

Once that this is done, do a PR and we then implemente the rest which is basically adding support for parsing such a unit definition from a string.

What do you think?

**Comment by hgrecco** [comment]
I am closing this, let's focus the discussion in #71
